specutils/fitting/spline_continuum.py

```diff
@@ -0,0 +1,485 @@
+        Output a model for the continuum
+
+    Raises
+    ------
+    ValueError
+       If: spectrum is not the correct type,
+       the exclude regions do not satisfy a list of sorted tuples,
+       the model and/or fitter are of the wrong type,
+
+    Examples
+    --------
+    TODO: add more and unit tests
+
+    See https://github.com/spacetelescope/dat_pyinthesky/blob/master/pyinthesky_specutils_fitting.ipynb
+
+    """
+
+    # Parameter checks
+    if not isinstance(spectrum, Spectrum1D):
+        raise ValueError('The spectrum parameter must be a Spectrum1D object')
+
+    exclude_regions = [] if exclude_regions is None else exclude_regions
+
+    for exclude_region in exclude_regions:
+        if len(exclude_region) != 2:
+            raise ValueError('All exclusion regions must be of length 2')
+        if exclude_region[0] >= exclude_region[1]:
+            raise ValueError('All exclusion regions must be (low, high)')
+
+    # Set default model and fitter
+    if model is None:
+        logging.info("Using Chebyshev1D(3) as default continuum model")
+        model = models.Chebyshev1D(3)
+
+    if fitter is None:
+        fitter = fitting.LevMarLSQFitter()
+
+    if not isinstance(model, modeling.FittableModel):
+        raise ValueError('The model parameter must be a astropy.modeling.FittableModel object')
+    # TODO: this is waiting on a refactor in modeling.fitting to work
+
+    # if not isinstance(fitter, fitting.Fitter):
+    #     raise ValueError("The model parameter must be an "
+    #                      "astropy.modeling.fitting.Fitter object.")
+
+    # Get input spectrum data
+    x = spectrum.spectral_axis.value
+    y = spectrum.flux.value
+
+    # Set up valid pixels mask. Exclude non-finite values.
+    good = np.isfinite(y)
+
+    # Exclude regions
+    for (excl1, excl2) in exclude_regions:
+        good[np.logical_and(x > excl1, x < excl2)] = False
+
+    # Set up sigma clipping
+    if sigma_lower is None:
+        sigma_lower = sigma
+
+    if sigma_upper is None:
+        sigma_upper = sigma
+
+    # Set the model as the default continuum in cases where the sigma
+    # clipping iterations == 0
+    continuum_model = model
+
+    for i_iter in range(iters):
+        logging.info("Iter {}: Fitting {}/{} pixels".format(
+            i_iter, good.sum(), len(good)))
+
+        # Fit model
+        # TODO: include data uncertainties
+        continuum_model = fitter(model, x[good], y[good])
+
+        # Sigma clip
+        difference = continuum_model(x) - y
+        finite = np.isfinite(difference)
+        sigma_difference = difference / np.std(difference[np.logical_and(good, finite)])
+        good[sigma_difference > sigma_upper] = False
+        good[sigma_difference < -sigma_lower] = False
+
+    if full_output:
+        return continuum_model, good
+
+    return continuum_model
+
+
+def fit_continuum_linetools(spec, edges=None, ax=None, debug=False, kind="QSO", **kwargs):
```

Comment:
@alexji can you expand on where this is planned to be used?

Reply:
There was not a plan, but it was requested in #231 so I ported it in.